src/NewProject/add_constraints_form.cpp

```diff
@@ -34,3 +45,11 @@ bool addConstraintsForm::IsCopySource() {
   return ui->m_ckkBoxCopy->checkState() == Qt::CheckState::Checked ? true
                                                                    : false;
 }
+
+void addConstraintsForm::pinAssign_opt_listen() {
+  Compiler *compiler = GlobalSession->GetCompiler();
+  if (!ui->select_random->isChecked())
```

Comment:
please change to if (ui->select_random->isChecked()). This is more ease to read.